src/Microsoft.AspNet.FileSystems/Implementation/PhysicalFileSystemWatcher.cs

```diff
@@ -35,12 +34,16 @@ internal PhysicalFileSystemWatcher(string root)
         internal IExpirationTrigger CreateFileChangeTrigger(string filter)
         {
             filter = NormalizeFilter(filter);
-            var pattern = WildcardToRegexPattern(filter);
+            var isWildCardSearch = IsWildCardSearch(filter);
```

Comment:
Can't we just implement checking `if(wildcard)` within the `FileChangeTrigger` itself so that it becomes an internal implementation detail rather than having separate trigger classes for wild card and regular file paths. Today files not monitored periodically for patterns is just a drawback which may eventually need to be fixed to provide guarantee. When that happens pattern / plain file name are not 2 different things. I feel that will also greatly simply the code changes needed to achieve this.